code/datums/components/harvester.dm

```diff
@@ -158,4 +158,8 @@
 ///Signal handler calling when user attacks
 /datum/component/harvester/proc/attack(datum/source, mob/living/target, mob/living/user, obj/item/weapon)
 	SIGNAL_HANDLER
+	var/datum/reagent/R = loaded_reagent
 	INVOKE_ASYNC(src, .proc/attack_async, source, target, user, weapon)
```

Comment:
why is this async again?

Reply:
because attack_async has do_after inside